src/libs/tools/timeline.c

```diff
@@ -616,12 +616,11 @@ static gboolean _time_read_bounds_from_collection(dt_lib_module_t *self)
   dt_lib_timeline_t *strip = (dt_lib_timeline_t *)self->data;
 
   sqlite3_stmt *stmt;
-  const char *query = "SELECT SUBSTR(db.datetime_taken, 1, 19) AS dt "
+  const char *query0 = "SELECT MIN(db.datetime_taken), MAX(db.datetime_taken) "
                       "FROM main.images AS db, memory.collected_images AS col "
-                      "WHERE db.id=col.imgid AND LENGTH(dt) = 19 AND"
```

Comment:
The problem is the mix of 20 (old ones) and 24 (new ones) characters long dates.
This should work if the database is updated with only 24 characters long dates (with a new db version) what I have avoided. That may have been a wrong choice.

Reply:
Shouldn't it work for either with `LENGTH(dt) >= 19` ?

I had initially tried using `SELECT MIN(dt), MAX(dt)`  from the existing SELECT but got an sql error...

Reply:
Yes, but it's noticeably slower.  String comparison for time will work as expected even in the presence of truncated strings, however:  
XX:03.999 < XX:04 < XX:04.000 < XX:04.001
And `_time_read_bounds_from_collection` only parses down to minutes anyway.

Reply:
> XX:03.999 < XX:04 < XX:04.000 < XX:04.001

Well, I'm surprised, but I've tested with sqlitestudio, and that works as you say. 
The exif datetime format (yyyy:mm:dd) is not one that sqlite doc mentions as a valid format but it seems to recognize this is a datetime field anyway ... Sorry for the noise.
That would also mean that it may not be necessary, in timeline.c, to check the length of string...